### PR TITLE
Use Epochs to load CoLA

### PR DESCRIPTION
**Pull request by sgugger**
This PR changes the way CoLA is loaded to use the Epochs API.
It removes the test set (that was unused) and does the minimal amount of changes to work with the script in BertCoLA.

Two notes where the design changed:
- renamed dev to validation as it's more expressive
- put the closure at the end of the init of the CoLA dataset so that we can use a trailing closure.

Added functionality: we can pass a random generator and thus get reproducible results.

### CODE REVIEW

**Comment by texasmichelle** [comment]
The kokoro failure is looking for the functions defined at the bottom of DataUtilities.swift. @BradLarson do we have a common function to use instead?

**Comment by sgugger** [comment]
Yeah I just realized that, I deleted the two files thinking they were useless now but did not try to build after, sorry about that. I can leave DatasetUtilities with them for now, there is a TODO to rely on the swift-models at some point, when it is done, this can be removed.

File: Examples/BERT-CoLA/main.swift [hunk 1]
```diff
@@ -41,21 +41,17 @@ var bertClassifier = BERTClassifier(bert: bert, classCount: 1)
...
-    exampleMap: exampleMapFn,
-    taskDirectoryURL: workspaceURL,
-    maxSequenceLength: maxSequenceLength,
-    batchSize: batchSize,
-    dropRemainder: true)
+  taskDirectoryURL: workspaceURL,
+  maxSequenceLength: maxSequenceLength,
+  batchSize: batchSize,
+  entropy: SystemRandomNumberGenerator()
+) { (example: CoLAExample) -> LabeledTextBatch in
```
Comment (texasmichelle) [inline Examples/BERT-CoLA/main.swift:49]: What are the advantages of moving to a trailing closure?

Comment (sgugger) [reply Examples/BERT-CoLA/main.swift:49]: It's more swift-idiomatic from what Dave constantly tells me ;)

The main advantage is that you don't have to define your `exampleMap` function separately and pass it, you can just put in curly brackets at the end (like in the main colab example).

**Comment by texasmichelle** [comment]
This is so helpful!! It cleans up the CoLA implementation considerably. Although there are aspects of `Epochs` that are not entirely clear to me at first glance, it's great that it covers so much of this functionality. Thank you!

**Comment by sgugger** [comment]
Thanks! Hope the design review on Friday will clarify the aspects of Epochs that are unclear.

**Comment by texasmichelle** [comment]
Created issues #488 and #489 for later follow-up.

**Comment by texasmichelle** [comment]
Fixes #480

**Pull request merged**